GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -31,22 +41,51 @@ static HttpRequestor()
             availableConnections = new SemaphoreSlim(ServicePointManager.DefaultConnectionLimit);
         }
 
-        public HttpRequestor(ITracer tracer, RetryConfig retryConfig, GitAuthentication authentication)
+        protected HttpRequestor(ITracer tracer, RetryConfig retryConfig, Enlistment enlistment)
         {
-            this.client = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true });
-            this.client.Timeout = retryConfig.Timeout;
             this.RetryConfig = retryConfig;
-            this.authentication = authentication;
+
+            this.authentication = enlistment.Authentication;
 
             this.Tracer = tracer;
 
+            var httpClientHandler = new HttpClientHandler() { UseDefaultCredentials = true };
+            httpClientHandler.ClientCertificateOptions = ClientCertificateOption.Manual;

```

Comment:
I don't think we should change anything about the mainline flow. Can we only set this when there is a cert provided?

Reply:
Moved the ClienCertificateOption assignment further down, to happen just before setting the certificate. The httpClientHandler isn't really new - it was just being created inline, in the HttpClient constructor.